### PR TITLE
refactor: v0 account model

### PR DESCRIPTION
**Pull request by enitrat**
# Pull Request type

Time spent on this PR: 0.6d

Resolves: #687

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: crates/ef-testing/src/evm_sequencer/evm_state/v0.rs [hunk 1]
```diff
@@ -33,13 +37,13 @@ impl Evm for KakarotSequencer {
         // Set the coinbase address.
         self.state_mut().set_storage_at(
             kakarot_address,
-            get_storage_var_address("coinbase", &[]),
+            get_storage_var_address("Kakarot_coinbase", &[]),
```
Comment (greged93) [inline crates/ef-testing/src/evm_sequencer/evm_state/v0.rs:40]: In constants as well?

File: crates/ef-testing/src/evm_sequencer/evm_state/v0.rs [hunk 2]
```diff
@@ -33,13 +37,13 @@ impl Evm for KakarotSequencer {
...
             coinbase_address.into(),
         )?;
 
         // Set the base fee.
         let [low_fee, high_fee] = split_u256(base_fee);
-        let basefee_address = get_storage_var_address("base_fee", &[]);
+        let basefee_address = get_storage_var_address("Kakarot_base_fee", &[]);
```
Comment (greged93) [inline crates/ef-testing/src/evm_sequencer/evm_state/v0.rs:46]: In constants as well?

File: crates/ef-testing/src/evm_sequencer/evm_state/v0.rs [hunk 3]
```diff
@@ -68,7 +72,7 @@ impl Evm for KakarotSequencer {
         // Set the block gas limit, considering it fits in a felt.
         let [block_gas_limit, _] = split_u256(block_gas_limit);
         let block_gas_limit = StarkFelt::from(block_gas_limit);
-        let block_gas_limit_address = get_storage_var_address("block_gas_limit", &[]);
+        let block_gas_limit_address = get_storage_var_address("Kakarot_block_gas_limit", &[]);
```
Comment (greged93) [inline crates/ef-testing/src/evm_sequencer/evm_state/v0.rs:75]: In constants as well?

File: crates/ef-testing/src/evm_sequencer/evm_state/v0.rs [hunk 4]
```diff
@@ -86,48 +90,38 @@ impl Evm for KakarotSequencer {
...
         }
 
-        // Set the Kakarot address.
-        storage.push(starknet_storage!(
-            "kakarot_address",
-            *self.environment.kakarot_address.0.key()
-        ));
+        storage.append(&mut vec![
+            starknet_storage!(ACCOUNT_IMPLEMENTATION, self.environment.eoa_class_hash.0), // both EOA and CA CH are the same (for now)
+            starknet_storage!("Ownable_owner", *self.environment.kakarot_address.0.key()),
```
Comment (greged93) [inline crates/ef-testing/src/evm_sequencer/evm_state/v0.rs:101]: In constants as well?

File: crates/ef-testing/src/evm_sequencer/evm_state/v0.rs [hunk 5]
```diff
@@ -86,48 +90,38 @@ impl Evm for KakarotSequencer {
...
+            starknet_address,
+            self.environment.contract_account_class_hash,
+        )?;
 
         // Add the address to the Kakarot evm to starknet mapping
         let kakarot_address = self.environment.kakarot_address;
         self.state_mut().set_storage_at(
             kakarot_address,
-            get_storage_var_address("evm_to_starknet_address", &[account.evm_address]),
+            get_storage_var_address("Kakarot_evm_to_starknet_address", &[account.evm_address]),
```
Comment (greged93) [inline crates/ef-testing/src/evm_sequencer/evm_state/v0.rs:124]: In constants as well?

File: crates/ef-testing/src/evm_sequencer/sequencer/v0.rs [hunk 6]
```diff
@@ -23,12 +23,11 @@ lazy_static! {
 
         let storage = [
             ("Ownable_owner", *KAKAROT_OWNER_ADDRESS.0.key()),
-            ("native_token_address", *ETH_FEE_TOKEN_ADDRESS.0.key()),
-            ("contract_account_class_hash", CONTRACT_ACCOUNT_CLASS_HASH.0),
-            ("externally_owned_account_class_hash", EOA_CLASS_HASH.0),
-            ("account_proxy_class_hash", PROXY_CLASS_HASH.0),
-            ("precompiles_class_hash", PRECOMPILES_CLASS_HASH.0),
-            ("block_gas_limit", StarkFelt::from(BLOCK_GAS_LIMIT))
+            ("Kakarot_native_token_address", *ETH_FEE_TOKEN_ADDRESS.0.key()),
```
Comment (greged93) [inline crates/ef-testing/src/evm_sequencer/sequencer/v0.rs:26]: Should as these be moved to a constants file?

**Review by greged93 — COMMENTED** [review]
Couple of nits. Could you also explain what the idea is behind this update? Are we dropping the proxy?

File: crates/ef-testing/src/evm_sequencer/evm_state/v0.rs (hunk 1 above)
Comment (enitrat) [reply crates/ef-testing/src/evm_sequencer/evm_state/v0.rs:40]: yeah definitely 👍

**Comment by enitrat** [comment]
yeah as per #687: 

Refactor the account setup flow to be compatible with the update of Kakarot0. Namely

No more proxy: we use the replace_class syscall
Renaming of storage variables with best practices
Remove distinction between EOAs and CAs in contracts

File: crates/ef-testing/src/evm_sequencer/sequencer/v0.rs (hunk 6 above)
Comment (enitrat) [reply crates/ef-testing/src/evm_sequencer/sequencer/v0.rs:26]: I think it can stay there but just use the constants

**Review by greged93 — APPROVED** [review]

**Pull request merged**